### DISCUSSION TITLE
Secret Manager and Socket.IO

### DISCUSSION THREAD

**Discussion by ghost** [googleapis/google-cloud-cpp#12673] [Q&A] [answered]
I am writing a program that is using Secret Manager from the SDK.  It also uses Socket.IO 
Ubuntu 22
SDK built manually (v2.15.1)
grpc built manually (v1.58.1)
Socket.IO client built manually (https://github.com/socketio/socket.io-client-cpp.git on the master branch, pulled on Oct 10, 2022):

My problem appears to be tied to SSL.  Socket.IO was built first and is using the system installed libssl (`/usr/lib/x86_64-linux-gnu/libssl.so`). Running `git submodule update --init` for gRPC brings down it's own SSL (`[submodule "third_party/boringssl-with-bazel"]`).  So, when I link, two different libssl are included:
`/usr/bin/c++ -g CMakeFiles/noThreadSIO.dir/noThreadSIO.cpp.o -o noThreadSIO  /usr/local/lib/libgoogle_cloud_cpp_secretmanager.a /usr/local/lib/libsioclient_tls.a /usr/lib/x86_64-linux-gnu/libssl.so /usr/local/lib/libssl.a  ` (snip)

If I remove the system libssl, linking fails due to missing functions.  Any ideas here?  Is my only option to rebuild gPRC without the `git submodule update`?

A test program that reproduces the segfault

```cc
#include <iostream>
#include "dotenv.h"
#include <sio_client.h>
#include <google/cloud/secretmanager/v1/secret_manager_client.h>

namespace secretmanager = ::google::cloud::secretmanager_v1;
static std::mutex smMutex;
static secretmanager::SecretManagerServiceClient *client = nullptr;

std::string getSecret()
{
    std::unique_lock<std::mutex> smlock(smMutex);
    std::stringstream secretKey;
    std::string result;

    if (client == nullptr)
    {
         client = new secretmanager::SecretManagerServiceClient(secretmanager::MakeSecretManagerServiceConnection());
    }

    secretKey << "projects/" << std::getenv("SECRET_PROJECT") << "/secrets/" << std::getenv("ENV") << '-' << std::getenv("APP") << "-" << std::getenv("PARM");
    std::cout << "Looking up " << secretKey.str() << std::endl;
    auto secret = client->GetSecret(secretKey.str());
    if (! secret)
    {
         std::cerr << "Failed to find secret key " << secretKey.str() << " : " << std::move(secret).status() << std::endl;
    }
    else
    {
        secretKey << "/versions/latest";
        auto request = google::cloud::secretmanager::v1::AccessSecretVersionRequest();
        request.set_name(secretKey.str());
        auto value = client->AccessSecretVersion(request);
        if (! value)
        {
            std::cerr << "Failed to find secret value " << secretKey.str() << " : " << std::move(secret).status() << std::endl;
        }
        else if (value->has_payload())
        {
            result = value->payload().data();
        }
        else
        {
            std::cerr << "No payload for secret value " << secretKey.str() << " : " << std::move(secret).status() << std::endl;
        }
    }
    smlock.unlock();
    return result;
}

int main(int, char**)
{
    dotenv::init("/var/www/.env");

// Comment this to prevent seg fault
    sio::client client;

    std::string value = getSecret();

    std::cout << "Got '" << value << "'" << std::endl;
}
```

**Comment by coryan** [comment, marked as answer]
We have instructions in [/doc/packaging.md](https://github.com/googleapis/google-cloud-cpp/blob/main/doc/packaging.md) showing how to compile gRPC using the system's SSL library.

/FYI @veblush

**Reply by ghost** [reply]
That is exactly what I needed.  Thank you!  (I don't see an option to mark this as the answer)

**Reply by coryan** [reply]
TIL: Only discussions in the Q&A category have "answers".

**Comment by coryan** [comment]
Glad we could help. Closing.
